### DISCUSSION TITLE
Capability of building things in Flask's extension pop-up instead of Snap-specific app

### DISCUSSION THREAD

**Discussion by porco-rosso-j** [MetaMask/snaps#2042] [Ideas]
This is copy&past of the same question I asked in [discord](https://discord.com/channels/697535391594446898/697538943498977400/1184878198891225219)

I've seen a few Snap examples for non-EVM chains, such as dogecoin-snap and starknet-snap. Then I wonder if, in the future, developers will be able to modify/add things on Flask's extension pop-up, instead of building a Snap-specific app page. 

I'm curious about this as having to visit a snap web app each time a user performs basic wallet operations like send/receive,  adding/removing wallets, and changing settings doesn't really seem user-friendly, given you don't have to do so with the normal MM extension.

I assume that it wouldn't be straightforward as systems around addresses, keys, and chain-id in those chains are simply different from EVMs. But I'm just wondering if there is such a possibility or not

**Comment by TravCrypto** [comment]
Great idea @porco-rosso-j, thanks for posting!

**Comment by Montoya** [comment]
Hi @porco-rosso-j! Please have a look at our feature roadmap and scroll down to the section on "Snaps homepages" https://github.com/MetaMask/snaps/discussions/1980

We will be sharing more information about this feature + dynamic UI in January :)

**Reply by TravCrypto** [reply]
Thanks for the quick response @Montoya 😄

**Reply by porco-rosso-j** [reply]
This is great!! thanks for the quick response :)
